core/src/ai/ai.spec.ts

```diff
@@ -0,0 +1,178 @@
+import { Board } from "../board";
+import { GameState } from "../game";
+import { getMockFromJson } from "../mocks/getMock";
+import { getMissingTilesForPath, highlightMissingTilesForPath } from "./ai";
+
+describe("findNextMove", () => {
+  it("should find missing tiles for path", () => {
+    const { state, board }: { state: GameState; board: Board } =
+      getMockFromJson("games/almostPath.json");
+    const tiles = getMissingTilesForPath(state.currentPlayer, board);
+    expect(tiles).toEqual([
+      { symbol: "A", color: "yellow" },
+      { symbol: "D", color: "cyan" },
+    ]);
+  });
+
+  it("should highlight missing tiles for path", () => {
+    const { state, board }: { state: GameState; board: Board } =
+      getMockFromJson("games/almostPath.json");
+    const updatedBoard = highlightMissingTilesForPath(
+      state.currentPlayer,
+      board,
+    );
+    expect(updatedBoard).toEqual([
+      [
+        null,
+        null,
+        null,
+        { symbol: "A", color: "magenta", moveAllowed: undefined },
```

Comment:
I'm not sure that `moveAllowed` is the best variable naming for hightlighted tiles. Can you find a better name ?

Reply:
This is the name we have since we began to develop allowed moves features. Anyway, i'm not sure to use it to highlight potentials winning tiles, and this test is no longer usefull